src/Orchard/Environment/ThreadProvider.cs

```diff
@@ -0,0 +1,9 @@
+using System.Threading;
+
+namespace Orchard.Environment {
+    public class ThreadProvider : IThreadProvider {
```

Comment:
Why, because I made a choice you would have made differently? Perhaps it seems overkill to provide an abstraction for just accessing a managed thread ID, but it does greatly simplify testing code if one can control the thread ID using a mock. Now, I haven't written a unit test to test logic used with the thread ID, but I will, and then I will be happy I did this. :)